app/code/Magento/Eav/Model/ResourceModel/ReadHandler.php

```diff
@@ -125,21 +125,18 @@ public function execute($entityType, $entityData, $arguments = [])
                     $select->where(
                         $metadata->getEntityConnection()->quoteIdentifier($scope->getIdentifier()) . ' IN (?)',
                         $this->getContextVariables($scope)
-                    )->order('t.' . $scope->getIdentifier() . ' DESC');
+                    )->order('t.' . $scope->getIdentifier() . ' ASC');
                 }
-                $selects[] = $select;
+                $attrTablesValues = $connection->fetchAll($select);
+                $attributeValues = array_merge($attributeValues, $attrTablesValues);
```

Comment:
Dont use array merge in foreach it is very slow 

See: https://github.com/kalessil/phpinspectionsea/blob/master/docs/performance.md#slow-array-function-used-in-loop

Better soultion 
```php
 foreach ($attributeTables as $attributeTable) {
                $select = $connection->select()
                    ->from(
                        ['t' => $attributeTable],
                        ['value' => 't.value', 'attribute_id' => 't.attribute_id']
                    )
                    ->where($metadata->getLinkField() . ' = ?', $entityData[$metadata->getLinkField()]);
                foreach ($context as $scope) {
                    //TODO: if (in table exists context field)
                    $select->where(
                        $metadata->getEntityConnection()->quoteIdentifier($scope->getIdentifier()) . ' IN (?)',
                        $this->getContextVariables($scope)
                    )->order('t.' . $scope->getIdentifier() . ' ASC');
                }
                $attributeValues[]  = $connection->fetchAll($select);
            }

            $attributeValues = array_merge(...$attributeValues);
			foreach ($attributeValues as $attributeValue) {
            ....
```